client/my-sites/customize/main.jsx

```diff
@@ -115,10 +123,32 @@ class Customize extends React.Component {
 		return false;
 	};
 
+	getReturnUrl = async () => {
+		const returnUrl = new URLSearchParams( window.location.search ).get( 'return' );
+
+		if ( ! returnUrl ) {
+			return null;
+		}
+
+		try {
+			const response = await wpcom.req.get( '/me/validate-redirect', { redirect_url: returnUrl } );
```

Comment:
**TIL**

I see there's another, s[imilar implementation](https://github.com/Automattic/wp-calypso/blob/c372ebe/client/blocks/login/continue-as-user.jsx#L25). 

Potential abstraction candidate for later?